packages/repo-tools/src/commands/api-reports/sql-reports/runSqlExtraction.ts

```diff
@@ -0,0 +1,243 @@
+  pkgName: string,
+  knex: Knex,
+  options: SqlExtractionOptions,
+) {
+  const migrationDir = cliPaths.resolveTargetRoot(
+    targetDir,
+    'migrations',
+    migrationTarget,
+  );
+
+  const reportName =
+    migrationTarget === '.' ? pkgName : `${pkgName}/${migrationTarget}`;
+
+  console.log(`Generating SQL report for ${reportName}`);
+
+  const migrationsListResult = await knex.migrate.list({
+    directory: migrationDir,
+  });
+  const migrations: string[] = migrationsListResult[1].map(
+    (m: { file: string }) => m.file,
+  );
+
+  const schemaInfoBeforeMigration = new Map<string, SchemaInfo>();
+
+  for (const migration of migrations) {
+    const schemaInfo = await getPgSchemaInfo(knex);
+    schemaInfoBeforeMigration.set(migration, schemaInfo);
+
+    await knex.migrate.up({
+      directory: migrationDir,
+      name: migration,
+    });
+  }
+
+  const schemaInfo = await getPgSchemaInfo(knex);
+
+  let failedDownMigration: string | undefined = undefined;
+  for (const migration of migrations.toReversed()) {
+    await knex.migrate.down({
+      directory: migrationDir,
+      name: migration,
+    });
+    const after = await getPgSchemaInfo(knex);
+    const before = schemaInfoBeforeMigration.get(migration);
+    if (!before) {
+      throw new Error(`No previous result for migration ${migration}`);
+    }
+
+    const diff = justDiff(before, after);
+    if (diff.length !== 0) {
+      console.log(
+        `Migration ${migration} is not reversible: ${JSON.stringify(
+          diff,
+          null,
+          2,
+        )}`,
+      );
+      failedDownMigration = migration;
+      break;
+    }
+  }
+
+  const report = prettyReport(
+    generateSqlReport({
+      reportName,
+      failedDownMigration,
+      schemaInfo,
+    }),
+  );
+
+  const reportPath = cliPaths.resolveTargetRoot(
+    targetDir,
+    `report${migrationTarget === '.' ? '' : `-${migrationTarget}`}.sql.md`,
+  );
+  const existingReport = await fs.readFile(reportPath, 'utf8').catch(error => {
+    if (error.code === 'ENOENT') {
+      return undefined;
+    }
+    throw error;
+  });
+  if (existingReport !== report) {
+    if (options.isLocalBuild) {
+      console.warn(`SQL report changed for ${targetDir}`);
+      await fs.writeFile(reportPath, report);
+    } else {
+      logApiReportInstructions();
+
+      if (existingReport) {
+        console.log('');
+        console.log(
+          `The conflicting file is ${relativePath(
+            cliPaths.targetRoot,
+            reportPath,
+          )}, expecting the following content:`,
+        );
+        console.log('');
+
+        console.log(report);
+
+        logApiReportInstructions();
+      }
+      throw new Error(`Report ${reportPath} is out of date`);
+    }
+  }
+}
+
+function prettyReport(content: string): string {
+  try {
+    const prettier = require('prettier') as typeof import('prettier');
+
+    const config = prettier.resolveConfig.sync(cliPaths.targetRoot) ?? {};
```

Comment:
In the next versions of prettier I believe the sync version is removed. To lower future migration hurdles, if it's possible, consider using the async versions of its api

Reply:
I plan a followup to refactor the API reports code, prolly best to handle as part of that. This is copied from a different place but didn't wanna do the whole refactor in this PR.